internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -529,6 +596,20 @@ func (b *BeatsMonitor) injectMetricsInput(
 		},
 	}
 
+	if usingOtelRuntime(componentInfos) {
+		prometheusStream := b.getPrometheusStream(failureThreshold, metricsCollectionIntervalString)
+		inputs = append(inputs, map[string]interface{}{
+			idKey:        fmt.Sprintf("%s-collector", monitoringMetricsUnitID),
+			"name":       fmt.Sprintf("%s-collector", monitoringMetricsUnitID),
+			"type":       "prometheus/metrics",
```

Comment:
Using the Metricbeat prometheus module I think is faster, but this means we'll be using ECS instead of SemConv and the OTLP schema for data from the collector which is our eventual end goal.

Did you look at using the upstream prometheus receiver to do this instead? The data format will be different (OTLP) and you'll need to use processors to set the target data stream.

Reply:
The more I think about this the more I think we only have two ways to do this:

1. We use either of the prometheus/metrics module or the prometheus receiver, but we need to process the data so the field names are the same as what they are today without introducing any new data streams. This is collecting the data from a different place but keeping it compatible.
2. We use only the prometheus receiver and ship the data in OTLP format to a new datastream. This is our end goal and is a breaking change to monitoring so can't go in a patch release.

Given the timeline for this I think we have to go with approach one here and try to get the data to look like it does now for the queue and outputs.

Reply:
We need to prototype option 1 ASAP, because if we can transform the internal telemetry pieces we need to keep the elastic_agent dashboard mostly working as is with the existing ECS data we can take the time we need to polish presenting the underlying OTLP data directly to users.